DOCSP-31487: removed options for command() v6 (#737)
mongodb/docs-node: source/fundamentals/run-command.txt

For a full list of database commands and corresponding parameters, see
the :ref:`Additional Information section <addl-info-runcommand>`.

.. _node-command-options:

Command Options
---------------

> You can set options when executing a database command by
> passing an options object to either command method. The ``command()`` method
> takes a ``RunCommandOptions`` object, and the ``runCursorCommand()`` method
> takes a ``RunCursorCommandOptions`` object.

Reviewer: Suggestion:

1) I think it would be more clear to specify the command methods rather than expecting the reader to associate it with content from the middle of the previous section.

2) I think "set option ... by passing an options object" is redundant.

3) I think the association between "options" object and the two types of options objects could be more clear.

```suggestion
You can specify optional command behavior for either the ``command()`` or the ``runCursorCommand()`` method.

The ``command()`` method accepts a RunCommandOptions object. To learn more about the RunCommandOptions type, see the API documentation.

The ``runCursor() method`` accepts a ``runCursorCommand()`` object. To learn more about the RunCursorCommandOptions type, see the API documentation.
```

Starting in version 6.0 of the {+driver-short+}, you can pass only the

> following options directly to the ``command()`` method:

Reviewer: Question: Does this apply to the runCursorCommand() method or only command() method? In either case, I think it could be good to call out the limitations for both methods.

Author: the ticket only says that it is for the `command()` method and from the feature PR, it seems  that `runCursorCommand()` is unchanged.

- ``comment``
- ``enableUtf8Validation``
- ``raw``
- ``readPreference``
- ``session``

You can set command-specific options, including a write or read

> concern, in the command document that you pass to ``command()``. The

Reviewer: Suggestion:

1) I think the reader would benefit from a clear differentiation between the command options described in the first paragraph and "command-specific options".

I also think that it might not be clear which options can be specified in the command document, and whether it depends on the command.

To contrast this with the "only the following options" from the prior paragraph:
"You can specify additional commands on certain commands..." 

2) I think the mention of write or read concerns as options that you can specify in the command document, followed by an example of maxTimeMS is confusing. To resolve this, provide the reader with a reference that they can check to determine which options they can provide which commands.

following code shows how to specify the ``maxTimeMS`` option in the
command document for a ``find`` command:

.. code-block:: javascript
   :emphasize-lines: 4

   const commandDoc = {
     find: "values",
     filter: { x: "abc" },
     maxTimeMS: 2000,
   };
   const result = await myDB.command(commandDoc);

.. note:: Read Preference

>    The ``command()`` and ``runCursorCommand()`` methods do not obey the read

Reviewer: Suggestion:

I think it would be good to avoid humanizing inanimate objects when possible (Use effective verbs from the style guide).

"The ... methods ignore" or "The ... methods throw an error if..."

Also, in the next sentence, I think "earlier" is more correct than "elsewhere".

For a full list of database commands and corresponding parameters, see
the :ref:`Additional Information section <addl-info-runcommand>`.

.. _node-command-options:

Command Options
---------------

You can specify optional command behavior for the ``command()``
and ``runCursorCommand()`` methods.

The ``command()`` method accepts a ``RunCommandOptions`` object. To learn
more about the ``RunCommandOptions`` type, see the `API documentation <{+api+}/types/RunCommandOptions.html>`__.

The ``runCursorCommand() method`` accepts a ``RunCursorCommandOptions``
object. To learn more about the ``RunCursorCommandOptions`` type, see
the `API documentation <{+api+}/types/RunCursorCommandOptions.html>`__.

Starting in version 6.0 of the {+driver-short+}, you can pass only the
following options directly to the ``command()`` method:

- ``comment``
- ``enableUtf8Validation``
- ``raw``
- ``readPreference``
- ``session``

You can set additional options, including a write or read
concern, in the command document that you pass to
the ``command()`` method. See the Server manual :manual:`entry

> </reference/command/>` for a command to learn about its associated

Reviewer: Issues:

1) I think this could be confusing to a user since the page this links to shows the overview of runCommand() and adminCommand() rather than command() and "its associated options". I also wasn't able to locate the aforementioned write or read concern options.

2) The text of the link should be descriptive. See [Link and Cross-Reference Examples](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/#link-and-cross-reference-examples).

Suggestion:

From looking at docs/manual/reference/command, it looks like the options are listed only on the target pages of the links of each command. It could be helpful to inform the reader that is how it is structured to reduce the chance of confusion. Also, I think it would be good to continue using the format "\<Page title\>" in the Server Manual.

E.g.
"Learn more about the available options for each command by following the link associated with the command in the :manual:\`User Commands </reference/command/#user-commands>\` section of the Server Manual"

Author: since the page describes commands outside of the User Commands category (such as `hello`), I will change the link and link text slightly

Reviewer: Resolving and adding a comment to the revised version.

options.

The following code shows how to specify a write concern
in the command document for a ``delete`` command:

.. code-block:: javascript
   :emphasize-lines: 4

   const commandDoc = {
       delete: "values",
       deletes: [ { q: { x: "abc" }, limit: 0 } ],
       writeConcern: { w: "majority", wtimeout: 5000 }
   };
   const result = await myDB.command(commandDoc);

.. note:: Read Preference

>    The ``command()`` and ``runCursorCommand()`` methods do not execute with
>    the read preference you may have set on your ``Db`` object earlier in

Reviewer: Suggestion:

I think it would be easier to understand and more concise by using [Positive Statements](https://www.mongodb.com/docs/meta/style-guide/writing/use-positive-statements/):

```suggestion
   The ``command()`` and ``runCursorCommand()`` methods ignore the
   read preference setting you may have set on your ``Db`` object.
```

I don't think it's necessary to mention "earlier in your code" since "you may have set" refers to the recent past.

.. _node-command-options:

Command Options
---------------

You can specify optional command behavior for the ``command()``
and ``runCursorCommand()`` methods.

The ``command()`` method accepts a ``RunCommandOptions`` object. To learn
more about the ``RunCommandOptions`` type, see the `API documentation <{+api+}/types/RunCommandOptions.html>`__.

The ``runCursorCommand() method`` accepts a ``RunCursorCommandOptions``
object. To learn more about the ``RunCursorCommandOptions`` type, see
the `API documentation <{+api+}/types/RunCursorCommandOptions.html>`__.

Starting in version 6.0 of the {+driver-short+}, you can pass only the
following options to the ``command()`` method:

- ``comment``
- ``enableUtf8Validation``
- ``raw``
- ``readPreference``
- ``session``

You can set additional options, including a write or read
concern, in the command document that you pass to
the ``command()`` method. To learn more about the available options for
a command, select it from the list of commands on the :manual:`Database
Commands </reference/command/>` page of the Server manual.

See the Server manual :manual:`entry
</reference/command/>` for a command to learn about its associated

> options.

Reviewer: Question:
Did you mean to include this? If so, it repeats same the in the previous paragraph (style guide issue and confusing user experience) and should be removed.
